liquidity-pool-changes/src/agent.spec.js

```diff
@@ -0,0 +1,171 @@
+
+const {
+  provideHandleBlock,
+  provideInitialize,
+  getPreviousLiquidity,
+} = require('./agent');
+
+// axios mock test
+describe('mock axios GET requests', () => {
+  it('should call axios.get and return the mocked response for CoinGecko', async () => {
+    mockCoinGeckoResponse.data = { '0xtokenaddress': { usd: 1000 } };
+    const response = await axios.get('https://url.url');
+    expect(axios.get).toHaveBeenCalledTimes(1);
+    expect(response.data['0xtokenaddress'].usd).toEqual(1000);
+
+    // reset call count for next test
+    axios.get.mockClear();
+    expect(axios.get).toHaveBeenCalledTimes(0);
+  });
+});
+
+// handler tests
+describe('large liquidity pool change agent', () => {
+  describe('handleBlock', () => {
+    let initializeData;
+    let handleBlock;
+
+    beforeEach(async () => {
+      initializeData = {};
+
+      await provideInitialize(initializeData)();
+
+      handleBlock = provideHandleBlock(initializeData);
+      initializeData.provider = {
+        getBalance: jest.fn().mockResolvedValue(mockPoolBalance),
+      }; // mock provider
+    });
+
+    it('returns empty findings if liquidity change is below given threshold', async () => {
+      // mock coin gecko response data
+      mockCoinGeckoResponse.data = {};
+      mockCoinGeckoResponse.data[mockToken0Address.toLowerCase()] = { usd: 1 };
+      mockCoinGeckoResponse.data[mockToken1Address.toLowerCase()] = {
+        usd: 2000,
+      };
+
+      const findings = await handleBlock();
+
+      expect(findings).toStrictEqual([]);
+      expect(mockPoolContract.token0).toHaveBeenCalledTimes(1);
+      expect(mockPoolContract.token1).toHaveBeenCalledTimes(1);
+      expect(mockPoolContract.balanceOf).toHaveBeenCalledTimes(2);
+      expect(initializeData.provider.getBalance).toHaveBeenCalledTimes(1);
+      expect(axios.get).toHaveBeenCalledTimes(1);
+
+      axios.get.mockClear();
+      mockPoolContract.token0.mockClear();
+      mockPoolContract.token1.mockClear();
+      mockPoolContract.balanceOf.mockClear();
+      initializeData.provider.getBalance.mockClear();
+    });
+
+    it('returns a finding if liquidity change is above the given threshold', async () => {
+      // mock coin gecko response data
+      mockCoinGeckoResponse.data = {};
+      mockCoinGeckoResponse.data[mockToken0Address.toLowerCase()] = { usd: 1 };
+      mockCoinGeckoResponse.data[mockToken1Address.toLowerCase()] = {
+        usd: 2000,
+      };
+
+      // run handleBlock() to set previous liquidity value for the first time. Should be no findings
+      const findings0 = await handleBlock();
+      expect(findings0).toStrictEqual([]);
+
+      const prevLiquidity = getPreviousLiquidity();
+
+      // run again and the agent should pick up on the changes of liquidity
+      const findings = await handleBlock();
+
+      const currentLiquidity = getPreviousLiquidity();
+
+      const percentChange = ((currentLiquidity - prevLiquidity) / prevLiquidity) * 100;
```

Comment:
This is not quite the right approach, in my opinion.  Specifically,
- This test does not set the threshold, it goes with the threshold passed in by a configuration file.  If that configuration file changes, this test may fail.
- Calculating the `percentChange` in liquidity from the handler's calculated liquidity values skips checking the actual liquidity calculation.  Consider selecting token amounts and USD conversion values for this test that would make the previous and current liquidity values round numbers.  Then you could hard-code those liquidity values and/or the percent change to check your findings against.
- The math on this line should be dealing with BigNumber.js type variables, so the BigNumber.js methods should be used (i.e. minus, div, and times)
- The percentChange calculation should also use `absoluteValue`